idea/src/org/jetbrains/kotlin/idea/inspections/blockingCallsDetection/CoroutineNonBlockingContextChecker.kt

```diff
@@ -30,7 +32,7 @@ import org.jetbrains.kotlin.resolve.lazy.BodyResolveMode
 class CoroutineNonBlockingContextChecker : NonBlockingContextChecker {
 
     override fun isApplicable(file: PsiFile): Boolean {
-        val languageVersionSettings = file.project.getLanguageVersionSettings()
+        val languageVersionSettings = file.module?.languageVersionSettings ?: file.project.getLanguageVersionSettings()
```

Comment:
I think here you should write just `file.languageVersionSettings` without these hacks. If you look inside, you see it handles your case, as well as some other cases.